src/stores/user-store.js

```diff
@@ -52,24 +53,83 @@ class UserStore extends EventEmitter {
         }
     }
 
-    registerActions (action) {
-        if (action.ACTION_TYPE === ACTION_TYPE.addNewUser) {
-            const index = this._findUserIndexByName(action.user);
-            if (index === -1 || action.force) {
-                this._addNewUser(action.user);
-                this.emit(EVENT_TYPE.userAdded);
+    _findUser (user) {
+        return this._users.filter(anyUser => {
+            return (anyUser.name.toLowerCase().includes(user.name.toLowerCase()) &&
+                    anyUser.phone.toLowerCase().includes(user.phone.toLowerCase()) &&
+                    anyUser.email.toLowerCase().includes(user.email.toLowerCase()) &&
+                    anyUser.website.toLowerCase().includes(user.website.toLowerCase())
+            );
+        });
+    }
+
+    _defineSearchFields () {
+        return {
+            name:    this._searchedUser.name !== '',
+            phone:   this._searchedUser.phone !== '',
+            email:   this._searchedUser.email !== '',
+            website: this._searchedUser.website !== ''
+        };
+    }
+
+    _actionAddNewUser (user, force) {
+        const index = this._findUserIndexByName(user);
+
+        if (index === -1 || force) {
+            this._addNewUser(user);
+            if (this._searchedUser instanceof UserInfo) {
+                const _foundUsers = this._findUser(this._searchedUser);
+
+                this.emit(EVENT_TYPE.usersFound, _foundUsers, this._defineSearchFields());
             }
             else {
-                const userId = this._users[index].id;
-                this.emit(EVENT_TYPE.addingFailed, userId);
+                this.emit(EVENT_TYPE.userAdded);
             }
         }
-        else if (action.ACTION_TYPE === ACTION_TYPE.updateUser) {
-            this._updateUser(action.user);
+        else {
+            const userId = this._users[index].id;
+            this.emit(EVENT_TYPE.addingFailed, userId);
+        }
+    }
+
+    _actionUpdateUser (user) {
+        this._updateUser(user);
+        if (this._searchedUser instanceof UserInfo) {
+            const _foundUsers = this._findUser(this._searchedUser);
+            this.emit(EVENT_TYPE.usersFound, _foundUsers, this._defineSearchFields());
+        }
+        else {
             this.emit(EVENT_TYPE.change);
         }
     }
 
+    _actionFindUser (user) {
+        this._searchedUser = user;
+        const _foundUsers  = this._findUser(user);
+        this.emit(EVENT_TYPE.usersFound, _foundUsers, this._defineSearchFields());
+    }
+
+    _actionStopFindUser () {
+        this._searchedUser = {};
+        this.emit(EVENT_TYPE.usersFound);
```

Comment:
We can avoid sending two events in a row.
We can reset all required state properies in the component on click on `Stop searching` button